src/robotFactory.js

```diff
@@ -1,17 +1,75 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
-}
+BaseRobot.prototype.goForward = function(step = 1) {
+  this.coords.y += step;
+};
+
+BaseRobot.prototype.goBack = function(step = 1) {
+  this.coords.y -= step;
+};
+
+BaseRobot.prototype.goRight = function(step = 1) {
+  this.coords.x += step;
+};
+
+BaseRobot.prototype.goLeft = function(step = 1) {
+  this.coords.x -= step;
+};
+
+BaseRobot.prototype.getInfo = function() {
+  return `Robot: ${this.name}, `
+  + `Chip version: ${this.chipVersion}, `
+  + `Weight: ${this.weight}`;
+};
+
+FlyingRobot.prototype = BaseRobot.prototype;
+
+function FlyingRobot(name, weight, coords, chipVersion) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
 
-function DeliveryDrone() {
-  // implement
+  this.coords.z = 0;
+};
+
+FlyingRobot.prototype.goUp = function(step = 1) {
+  this.coords.z += step;
+};
+
+FlyingRobot.prototype.goDown = function(step = 1) {
+  this.coords.z -= step;
+};
+
+DeliveryDrone.prototype = FlyingRobot.prototype;
+
+function DeliveryDrone(name,
+  weight,
+  coords,
+  chipVersion,
+  maxLoadWeight,
+  currentLoad) {
+  BaseRobot.call(this, name, weight, coords, chipVersion);
+  this.maxLoadWeight = maxLoadWeight;
+  this.currentLoad = currentLoad;
```

Comment:
разделяй код пустыми строками
```suggestion
  maxLoadWeight,
  currentLoad,
  ) {
  BaseRobot.call(this, name, weight, coords, chipVersion);
  
  this.maxLoadWeight = maxLoadWeight;
  this.currentLoad = currentLoad;
```